tests/ValidateLogicTest.php

```diff
@@ -147,7 +148,13 @@ public function getRules(): iterable
     public function testBase(array $expectedErrorMessages, mixed $data, iterable|object|string|null $rules): void
     {
         $result = $this->createValidator()->validate($data, $rules);
-        $this->assertSame($expectedErrorMessages, $result->getErrorMessagesIndexedByAttribute());
+        $this->assertEquals(
+            $expectedErrorMessages,
+            array_map(
+                fn (array $errors) => array_map(fn (ErrorMessage $error) => $error->getMessage(), $errors),
+                $result->getErrorMessagesIndexedByAttribute(),
+            )
+        );
```

Comment:
This is quite bad UX, I think.

Reply:
It returns all parts of `ErrorMessage` to satisfy user's needs if the one wants to expose raw value to a frontend.
Something like that. If you have better design decision describe it please.